src/controller/CHIPDeviceController.cpp

```diff
@@ -2372,6 +2374,37 @@ CHIP_ERROR DeviceCommissioner::ParseICDInfo(ReadCommissioningInfo & info)
         }
     }
 
+    if (!isICD)
+    {
+        info.icd.idleModeDuration    = 0;
+        info.icd.activeModeDuration  = 0;
+        info.icd.activeModeThreshold = 0;
+        return CHIP_NO_ERROR;
+    }
+
+    err = mAttributeCache->Get<IcdManagement::Attributes::IdleModeDuration::TypeInfo>(kRootEndpointId, info.icd.idleModeDuration);
+    if (err != CHIP_NO_ERROR)
```

Comment:
All these blocks could be simplified with the `VerifyOrReturnError` marcros. Up to you.

Reply:
ideally we can use VerifyOrReturnLogError, but it cannot print component, so I decide not to use macro. thanks